File: client-clojure/src/sirqul_io_t_platform/specs/coords_response.clj
(ns sirqul-io-t-platform.specs.coords-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.name-string-value-response :refer :all]
            )
  (:import (java.io File)))


(def coords-response-data
  {
   (ds/opt :valid) boolean?
   (ds/opt :message) string?
   (ds/opt :version) float?
   (ds/opt :serializeNulls) boolean?
   (ds/opt :startTimeLog) int?
   (ds/opt :errorCode) string?
   (ds/opt :request) (s/coll-of name-string-value-response-spec)
   (ds/opt :latitude) float?
   (ds/opt :longitude) float?
   (ds/opt :country) string?
   (ds/opt :countryCode) string?
   (ds/opt :region) string?
   (ds/opt :city) string?
   (ds/opt :postalCode) string?
   (ds/opt :regionCode) string?
   (ds/opt :returning) string?
   })

(def coords-response-spec
  (ds/spec
    {:name ::coords-response
     :spec coords-response-data}))

File: client-scalaz/src/main/scala/org/openapitools/client/api/CoordsResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import CoordsResponse._

case class CoordsResponse (
  valid: Option[Boolean],
message: Option[String],
version: Option[Double],
serializeNulls: Option[Boolean],
startTimeLog: Option[Long],
errorCode: Option[String],
request: Option[List[NameStringValueResponse]],
latitude: Option[Double],
longitude: Option[Double],
country: Option[String],
countryCode: Option[String],
region: Option[String],
city: Option[String],
postalCode: Option[String],
regionCode: Option[String],
returning: Option[String])

object CoordsResponse {
  import DateTimeCodecs._

  implicit val CoordsResponseCodecJson: CodecJson[CoordsResponse] = CodecJson.derive[CoordsResponse]
  implicit val CoordsResponseDecoder: EntityDecoder[CoordsResponse] = jsonOf[CoordsResponse]
  implicit val CoordsResponseEncoder: EntityEncoder[CoordsResponse] = jsonEncoderOf[CoordsResponse]
}
